controllers/registry_controller.go

```diff
@@ -94,7 +95,20 @@ func (r *registryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("unable to create signer from private key secret: %w", err)
 	}
-	for _, node := range nodes.Items {
+	for _, n := range nodes.Items {
+		// Ensure the node status is up to date, going through this list could take a long time depending on the
+		// number of Windows nodes in the cluster
+		var node core.Node
+		if err := r.client.Get(ctx, types.NamespacedName{Name: n.GetName()}, &node); err != nil {
+			if k8sapierrors.IsNotFound(err) {
+				continue
+			}
+			return ctrl.Result{}, fmt.Errorf("unable to get node %s: %w", n.GetName(), err)
+		}
+		if node.Status.Phase == core.NodeTerminated {
```

Comment:
The field is never populated, and now is deprecated.

https://github.com/openshift/windows-machine-config-operator/blob/master/vendor/k8s.io/api/core/v1/types.go#L6230

Reply:
https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/node-v1/#:~:text=phase%20(string)-,NodePhase,-is%20the%20recently